dev-client/src/components/projects/ProjectPreviewCard.tsx

```diff
@@ -1,56 +1,49 @@
-import {Badge, Box, HStack, Heading, Link, Text} from 'native-base';
-import {ProjectPreview} from '../../types';
+import {Box, HStack, Heading, Link, Text} from 'native-base';
 import {useTranslation} from 'react-i18next';
-import {useNavigation} from '@react-navigation/native';
-import {RootStackParamList, ScreenRoutes} from '../../screens/constants';
+import {ScreenRoutes} from '../../screens/constants';
 import {useCallback} from 'react';
-import {fetchProject} from '../../dataflow';
-import type {NativeStackNavigationProp} from '@react-navigation/native-stack';
+import {
+  Project,
+  fetchProject,
+} from 'terraso-client-shared/project/projectSlice';
+import {useNavigation} from '../../screens';
+import {useDispatch} from '../../model/store';
+import IconChip from '../common/IconChip';
 
 type Props = {
-  project: ProjectPreview;
+  project: Project;
 };
 
-// see https://reactnavigation.org/docs/typescript/#type-checking-screens
-type ProjectListProp = NativeStackNavigationProp<
-  RootStackParamList,
-  ScreenRoutes.PROJECT_LIST
->;
-
 export default function ProjectPreviewCard({project}: Props) {
   const {t} = useTranslation();
-  // see https://reactnavigation.org/docs/typescript/#annotating-usenavigation
-  // TODO: Wrap this in our own custom hook?
-  const navigation = useNavigation<ProjectListProp>();
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
 
-  const goToProject = useCallback(() => {
-    const p = fetchProject(project.id);
-    return navigation.navigate(ScreenRoutes.PROJECT_VIEW, {project: p});
-  }, [project, navigation]);
+  const goToProject = useCallback(async () => {
+    await dispatch(fetchProject(project.id));
+    return navigation.navigate(ScreenRoutes.PROJECT_VIEW);
```

Comment:
question (for a future PR): how should we handle this in general? this ensures the project data exists before navigating to the screen, but has the downside that we have to wait for a backend request to finish responding to the button press even if we already have the project data

Reply:
Yeah, that's a great point. I've been thinking a bit about how we should structure the global state. I think it makes sense to take another look at this when we starting adding offline stuff. I made another issue and tentatively linked it to the offline user story. I think we can take another look whenever the moment seems right to us.

Reply:
In the meantime, I think you're right and we should use a selector to check if this project data has already been updated.